### ISSUE TITLE
Fail2ban does not work on logging failed

### ISSUE THREAD

**Issue by paradoxisme** [YunoHost/issues#1672]
Hello,

I've tried to login into yunohost instance a lot of time with bad password with 4G phone. My ip adress is new baned.

Version : Yunohost 4.0.8
Tested on fresh install (lxc container with proxmox)

I've search which files are modified when login is correct and bad : 
The only file change is /var/log/nginx/WEB_SITE-access.log

Access_GRANTED : 
```
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:52:04 +0000] "POST /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv HTTP/2.0" 302 154 "https://WEB_SITE.fr/yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv" "Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile DuckDuckGo/5 Safari/537.36"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:52:04 +0000] "GET /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv HTTP/2.0" 302 154 "https://WEB_SITE.fr/yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv" "Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile DuckDuckGo/5 Safari/537.36"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:52:04 +0000] "GET / HTTP/2.0" 302 154 "https://WEB_SITE.fr/yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv" "Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile DuckDuckGo/5 Safari/537.36"
XXX.XXX.XXX.XXX - USER [31/Oct/2020:10:52:04 +0000] "GET /yunohost/sso/ HTTP/2.0" 200 3141 "https://WEB_SITE.fr/yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv" "Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile DuckDuckGo/5 Safari/537.36"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:52:05 +0000] "GET /favicon.ico HTTP/1.1" 302 154 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:52:05 +0000] "GET /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIvZmF2aWNvbi5pY28= HTTP/1.1" 200 3087 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:52:05 +0000] "GET /favicon.ico HTTP/1.1" 302 154 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:52:05 +0000] "GET /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIvZmF2aWNvbi5pY28= HTTP/1.1" 200 3087 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
```

Access_DENIED : 
```
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:51:28 +0000] "POST /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv HTTP/2.0" 302 154 "https://WEB_SITE.fr/yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv" "Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile DuckDuckGo/5 Safari/537.36"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:51:28 +0000] "GET /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv HTTP/2.0" 200 3117 "https://WEB_SITE.fr/yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv" "Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile DuckDuckGo/5 Safari/537.36"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:51:28 +0000] "GET /favicon.ico HTTP/1.1" 302 154 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:51:28 +0000] "GET /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIvZmF2aWNvbi5pY28= HTTP/1.1" 200 3087 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:51:28 +0000] "GET /favicon.ico HTTP/1.1" 302 154 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:51:29 +0000] "GET /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIvZmF2aWNvbi5pY28= HTTP/1.1" 200 3087 "-" "Dalvik/2.1.0 (Linux; U; Android 10; ONEPLUS A3003 Build/QQ3A.200805.001)"
```

But the pattern in /etc/fail2ban/filter.d/yunohost.conf is never found

**Comment by paradoxisme** [comment]
May be we can create pattern from this log : 
XXX.XXX.XXX.XXX - - [31/Oct/2020:10:51:28 +0000] "GET /yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv HTTP/2.0" 200 3117 "https://WEB_SITE.fr/yunohost/sso/?r=aHR0cHM6Ly9oeWRyb3guZnIv" "Mozilla/5.0 (Linux; Android 10) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/85.0.4183.101 Mobile DuckDuckGo/5 Safari/537.36"

**Comment by alexAubin** [comment]
Hi there,

thanks for your report

Indeed the fail2ban mechanism for SSO is broken and I was a bit puzzled why. Turns out that the fail2ban rule isn't supposed to be based on the POST request (and even less on GET request). It's supposed to find : 

```
helpers.lua:[0-9]+: authenticate\(\): Connection failed for: .*, client: <HOST>
```

in some nginx's `-error.log` file

But turns out that the piece of code that was logging the "Connection failed" got savagely removed by myself during The First Great War Of Permissions™ as we like to call it (or version 3.7 for short) because I absolutely did not realize this was used for this ...

https://github.com/YunoHost/SSOwat/commit/1eb322df1754ebb0a956216c5984e23d1caabb5a#

Soooo gonna push a commit to fix this, thanks !

**Comment by alexAubin** [comment]
Fixed by https://github.com/YunoHost/SSOwat/commit/6a7a9d668e5a23ad5d1c0a1630c7b2c2b82ef19c ... maybe will do a hotfix release...

**Comment by paradoxisme** [comment]
Ok thank a lot ;) !
I've tested your change and it's work. My phone is banned after 10 try.

**Comment by alexAubin** [comment]
Fix released in ssowat 4.0.5 stable
